### ISSUE TITLE
Change documented/recommended configuration style to the flat style.

### ISSUE THREAD

**Issue by thsig** [garden-io/garden#544]
This PR https://github.com/garden-io/garden/pull/535 added support for a "flat" configuration style that's closer to the YAML syntax for defining k8s .

To illustrate, here's a module definition using the old/nested style:
```yaml
module:
  description: Hello world
  type: container
  ...
```
and an equivalent definition using the new/flat style:
```yaml
kind: Module
description: Hello world
type: container
...
```
The flat style can also be used for projects (in which case `kind` would be `Project`).

While that PR was under review, we discussed making the flat style the default/recommended/documented configuration style, so I'm logging that idea as a Github issue — we may want to make a decision on this in the next few weeks.

**Comment by edvald** [comment]
I'd like to do this around the 0.10 release. @eysi09 what do you think?

**Comment by eysi09** [comment]
Yeah, sounds good.

**Comment by eysi09** [comment]
Closed with #775
